### ISSUE TITLE
booleanize Vector objects

### ISSUE THREAD

**Issue by JeffVenancius** [godotengine/godot-proposals#7475]
### Describe the project you are working on

A top-down jrpg.

### Describe the problem or limitation you are having in your project

Well, it's pretty simple, really.
As we use Vector2.ZERO (for example) a lot and we do that in conditions, it makes sense and it shorts some coding to convert them to booleans sometimes.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Currently you can do it implicitly, as long as the parameters are not set in stone. The idea is just a **bool(Vector2)** just so you can write statically but avoiding, you know, one if statement.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Statically is like this:
```
var move_dir := Vector2.ZERO
	if move_dir:
		set_walk(true)
	else:
		set_walk(false)
```

My idea is to be like this:
```
var move_dir := Vector2.ZERO
	set_walk(bool(move_dir))
```

### If this enhancement will not be used often, can it be worked around with a few lines of script?

it's a quality of life and I think it makes sense to say that we already do.

### Is there a reason why this should be core and not an add-on in the asset library?

Same reason why there's explicit conversions to other data types.

**Comment by KoBeWi** [comment]
You can already use a bit more explicit form: `bool(move_dir != Vector2())`.

**Comment by zinnschlag** [comment]
@KoBeWi What is the bool for? The value of a != comparison should already be a bool.

**Comment by KoBeWi** [comment]
Right, so you can do `set_walk(move_dir != Vector2())`

**Comment by JeffVenancius** [comment]
> You can already use a bit more explicit form: `bool(move_dir != Vector2())`.

Yeah, but it's a workaround, I think a more direct conversion would be more readable while still shortening the code.
Also, I think this way of doing it a little confusing, ` move_dir != Vector2() ` means what? I know it by experience, etc... But it's weird to have to think about it, you know?

**Comment by KoBeWi** [comment]
It's not like `bool(move_dir)` is straight-forward either. What is the result of this operation? `false` if `move_dir` is null? Or when it's `Vector2.ONE`? (like in Color's case) It still requires some extra knowledge.

**Comment by JeffVenancius** [comment]
> It's not like `bool(move_dir)` is straight-forward either. What is the result of this operation? `false` if `move_dir` is null? Or when it's `Vector2.ONE`? (like in Color's case) It still requires some extra knowledge.

Null is falsy, isn't?
Also, Color is the strange one then - I mean: zero is zero, not zero is not zero.

**Comment by KoBeWi** [comment]
> Null is falsy, isn't?

Yeah, but Vector2 can't be null if you use static typing.

**Comment by JeffVenancius** [comment]
Yeah but it won't matter(?)
I mean, if you say `bool(0)` will allways be false, there's no int with null value, is there?

**Comment by aaronfranke** [comment]
I am against this proposal, I think the code is more readable when you include the `!= Vector2.ZERO` or similar.

**Comment by timothyqiu** [comment]
You probably also want use `move_dir.is_zero_approx()` to prevent problems that may arise from floating-point calculation errors.

**Comment by aoisensi** [comment]
It feels unnatural that you can do this in `if` statements but not with `bool()`.
